Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -69,85 +172,217 @@ def highjack_transition_rando():
+    while new_area_reached:
+        new_area_reached = False
+        new_links_reached = [
+            x for x in unchecked_links if (x[0] in areas_reachable or x[1] in areas_reachable)
+        ]
+        if len(new_links_reached) > 0:
+            new_area_reached = True
+            for nl in new_links_reached:
+                if nl[0] not in areas_reachable:
+                    areas_reachable.append(nl[0])
+                elif nl[1] not in areas_reachable:
+                    areas_reachable.append(nl[1])
+                unchecked_links.remove(nl)
+    return len(areas_reachable) == len(area_list)
+
+
+def break_open_connection(
+    level_list: Sequence[Area],
+    index: int,
+    link_list: list[tuple[Area, Area]],
+):
+    global total_con_left
+    direc = random.choice([-1, 1])
+    link_i = random.randrange(len(link_list))
+    valid_link = False
+    while not valid_link:
+        linked_areas: list[Area] = []
+        for i in range(len(level_list)):
+            if i == index:
+                break
+            linked_areas.append(level_list[i])
+        valid_link = check_part_of_loop(link_list[link_i], link_list, linked_areas)
+        if not valid_link:
+            link_i += direc
+            if link_i == len(link_list):
+                link_i = 0
+            elif link_i < 0:
+                link_i += len(link_list)
+    level_a, level_b = link_list.pop(link_i)
+    unlink_two_levels(level_a, level_b)
+    total_con_left += 2
+
+
+def link_list_to_transitions(
+    link_list: list[tuple[Area, Area]],
+    transitions_map: MutableMapping[tuple[int, int], Transition],
+    origins_bucket: list[Transition],
+    redirections_bucket: list[Transition],
+):
+    for link in link_list:
+        options_original = [
+            trans for trans in origins_bucket
+            if trans.from_ == link[0].area_id
+        ]
+        options_redirect = [
+            trans for trans in redirections_bucket
+            if trans.to == link[1].area_id
+        ]
+        original = random.choice(options_original)
+        redirect = random.choice(options_redirect)
+        transitions_map[original] = redirect
+        origins_bucket.remove(original)
+        redirections_bucket.remove(redirect)
+
+        counterpart_original = Transition(from_=redirect.to, to=redirect.from_)
+        counterpart_redirect = Transition(from_=original.to, to=original.from_)
+        transitions_map[counterpart_original] = counterpart_redirect
+        origins_bucket.remove(counterpart_original)
+        redirections_bucket.remove(counterpart_redirect)
+
+
+def get_random_redirection(original: Transition, all_redirections: Iterable[Transition]):
+    possible_redirections = [
+        redirect for redirect in all_redirections
+        if original.from_ != redirect.to  # Prevent looping on itself
+    ]
+    if len(possible_redirections) > 0:
+        return random.choice(possible_redirections)
+    return None
 
 
 def set_transitions_map():
```

Comment:
The Ruff lint here is about this method being too complex. I was already thinking about breaking it up just looking at it but it seems it just hit the linter's configured complexity limits.

You can ignore it for now with:
```suggestion
def set_transitions_map():  # noqa: PLR0912, PLR0915 # TODO: Break up in smaller functions
```

Reply:
Done.